core/src/main/java/hudson/model/TextParameterDefinition.java

```diff
@@ -37,7 +38,7 @@ public TextParameterDefinition(String name, String defaultValue, String descript
         super(name, defaultValue, description);
     }
 
-    @Extension
+    @Extension @Symbol({"text","textParam"})
```

Comment:
This is because Job DSL plugins uses "xyzParam" names, but it's against what the naming guideline says --- it adds a suffix that refers to the extension point.

So I'm giving the primary name that matches the naming guideline `text` and giving it an alias that matches what Job DSL plugin uses.